drivers/flash/spi_nor.c

```diff
@@ -33,19 +69,58 @@ struct spi_nor_data {
 #ifdef DT_INST_0_JEDEC_SPI_NOR_CS_GPIOS_CONTROLLER
 	struct spi_cs_control cs_ctrl;
 #endif /* DT_INST_0_JEDEC_SPI_NOR_CS_GPIOS_CONTROLLER */
+#ifdef DT_INST_0_JEDEC_SPI_NOR_HAS_DPD
+	/* Low 32-bits of uptime counter at which device last entered
+	 * deep power-down.
+	 */
+	u32_t ts_enter_dpd;
+#endif
 	struct k_sem sem;
 };
 
-#if defined(CONFIG_MULTITHREADING)
```

Comment:
actually, this SYNC_* macros being replaced by new functions should be in its own commit. It is not related to deep power down.

Reply:
It sorta is, since it's the added complexity of handling DPD that makes the macros unwieldy.  However, if you really want it that way I can do it.